[MRG+1] Reduce whitespace around plots

soupault opened the pull request:
## References
Closes https://github.com/scikit-image/scikit-image/issues/1773 and, particularly, https://github.com/scikit-image/scikit-image/issues/1773#issuecomment-154585209.

**UPD 04.09.2016:** closes also https://github.com/scikit-image/scikit-image/issues/2280.

All modified examples have been manually verified.

sciunto commented:
I didn't compiled and checked thoroughly, but it's a nice commit :)

ahojnnes commented on doc/examples/edges/plot_edge_filter.py:
```diff
@@ -19,17 +19,20 @@
-ax0.set_title('Roberts Edge Detection')
-ax0.axis('off')
+ax[0].imshow(edge_roberts, cmap=plt.cm.gray)
+ax[0].set_title('Roberts Edge Detection')
```
the titles in this figure are way too small

ahojnnes commented on doc/examples/edges/plot_edge_filter.py:
```diff
@@ -64,23 +67,24 @@
-ax0.set_title('Original image')
-ax0.axis('off')
+ax[1].imshow(edge_scharr, cmap=plt.cm.gray)
+ax[1].set_title('Scharr Edge Detection')
```
also too small in this figure

ahojnnes commented on doc/examples/edges/plot_marching_cubes.py:
```diff
@@ -37,7 +37,7 @@
 # Display resulting triangular mesh using Matplotlib. This can also be done
 # with mayavi (see skimage.measure.marching_cubes docstring).
-fig = plt.figure(figsize=(10, 12))
+fig = plt.figure(figsize=(12, 12))
```
the axes are too small here

ahojnnes commented on doc/examples/filters/plot_rank_mean.py:
```diff
@@ -33,17 +32,17 @@
 
-
-fig, axes = plt.subplots(nrows=2, ncols=2, figsize=(8, 10),
+fig, axes = plt.subplots(nrows=2, ncols=2, figsize=(12, 12),
```
titles too small

ahojnnes commented on doc/examples/segmentation/plot_label.py:
```diff
@@ -39,19 +38,17 @@
 image_label_overlay = label2rgb(label_image, image=image)
 
-fig, ax = plt.subplots(ncols=1, nrows=1, figsize=(6, 6))
+fig, ax = plt.subplots(figsize=(12, 6))
```
Axes too small, but I suggest to remove them here entirely. There is no benefit in knowing the pixel coordinates here...

ahojnnes commented on doc/examples/xx_applications/plot_rank_filters.py:
```diff
@@ -44,11 +44,15 @@
 noisy_image = img_as_ubyte(data.camera())
```
For pretty much all figures here, the titles and axes are too small...

soupault replied:
@ahojnnes Thanks for review! Yes, I agree. Titles become too small in `sphinx-gallery`.
I'll give the PR an another round and will notify you on results.

sciunto commented:
@soupault Did you get free time to work on this?  (gentle ping ;) )

soupault commented:
@sciunto not yet, but I'm definitely planning to finalize this shortly.

sciunto commented:
cool!

soupault commented:
@ahojnnes I've reviewed and fixed all occurencies of tine titles, enhanced some layouts and addressed your other remarks. Feel free to revisit.

Fixed https://github.com/scikit-image/scikit-image/issues/2280 as well.

/cc @sciunto @emmanuelle

sciunto commented on doc/examples/features_detection/plot_blob.py:
```diff
@@ -36,13 +36,14 @@
-from math import sqrt
 from skimage.color import rgb2gray
 
 import matplotlib.pyplot as plt
```
Why moving math on the top and not this one? if we do external and then internal...

soupault replied:
I've worked out a convention for myself to split the imports into several blocks in the following order:

- standard modules shipped within Python distribution (`math`, `argparse`, `os`, etc);
- PyData stack (in order: `numpy`, `scipy`, `matplotlib`, `skimage`, etc);
- other trusted modules (`pydy`, `nipy`, whatever);
- thrid party modules: `opencv`, `tensorflow`, etc.

IMO, this order makes more sense. Let me know if you insist to change this, I don't mind.

sciunto replied:
ok. I was considering skimage as internal, but indeed, it's an example...

ahojnnes commented:
LGTM, MRG+1.

sciunto commented on doc/examples/segmentation/plot_label.py:
```diff
@@ -39,19 +37,18 @@
 
 for region in regionprops(label_image):
-
     # skip small images
```
Should we update this comment then. Like "Keep only large areas".

sciunto commented on doc/examples/segmentation/plot_ncut.py:
```diff
@@ -26,8 +27,13 @@
-plt.figure()
-io.imshow(out2)
-io.show()
+fig, ax = plt.subplots(nrows=2, sharex=True, sharey=True,
```
single line? looks < 80 chars.

sciunto commented on doc/examples/segmentation/plot_rag_draw.py:
```diff
@@ -15,18 +16,21 @@
 labels = segmentation.slic(img, compactness=30, n_segments=400)
 g = graph.rag_mean_color(img, labels)
 
+fig, ax = plt.subplots(nrows=2, sharex=True, sharey=True,
```
single line?

sciunto commented on doc/examples/xx_applications/plot_rank_filters.py:
```diff
@@ -32,7 +32,6 @@
 
 .. [1] Pierre Soille, On morphological operators based on rank filters, Pattern
        Recognition 35 (2002) 527-535.
```
Since you are here, would you like to add `DOI:10.1016/S0031-3203(01)00047-4`

sciunto commented:
Nice improvements. I didn't compile the doc, I just left few comments while proofreading.

soupault commented:
@sciunto thanks, all addressed!

sciunto commented:
@ahojnnes If you check the rendering, I'm +1.

sciunto commented:
I have several broken plots from your branch
* plot_segmentations
* plot_marching_cubes
* plot_rag_draw
* plot_skeletonize

Is it just me? I ran a make clean before...

soupault commented:
@sciunto Just checked out once again with a clean build - everything works fine for me.
`plot_skeletonize` -> you mean `plot_skeleton.py`?

soupault commented:
@sciunto wait, those examples have underying functions updated recently. Have you run `python setup.py develop` first?

sciunto commented:
OK, looks like it's just me. Few imports seem to fail for a reason I don't get. Anyway. I'm merging :)

Thanks @soupault

sciunto commented:
damn... i was not in my virtualenv... stupid.

sciunto commented:
@soupault Do you have this for plot_ncut?

![sphx_glr_plot_ncut_001](https://cloud.githubusercontent.com/assets/335370/18233660/5bf6feca-72ed-11e6-9b3d-10b877f76b59.png)

sciunto commented:
Some titles look really small as @ahojnnes  mentioned at some point. Like
![sphx_glr_plot_line_hough_transform_001](https://cloud.githubusercontent.com/assets/335370/18233689/20e220ac-72ee-11e6-9768-91329b85484a.png)

sciunto commented:
Titles on plot wathershed are cut. Sorry, I'm a bit difficult.

soupault commented:
@sciunto 
Regarding `plot_ncut.py`: I see the same image both for old (so I cannot even reproduce our current example on the website with the same code) and new version of source code. I have no idea what is this related to, actually. Maybe the newer version of `matplotlib` causes this. There is a lonely clue:
```
loading pickled environment... done
Executing file ../examples/segmentation/plot_ncut.py
/home/egor/Workspace/soupault-scikit-image/skimage/io/_plugins/matplotlib_plugin.py:74: UserWarning: Low image dynamic range; displaying image with stretched contrast.
  warn("Low image dynamic range; displaying image with "
../examples/segmentation/plot_ncut.py ran in : 3.9 seconds
```
Let's keep it for the later investigation.

As for `plot_line_hough_transform.py` example: it is really hard to adjust those particular plots. For some reason, `tight_layout` does work for that example at all (maybe due to `extend` parameter), so the only instrument to work with is `figsize`. So we can either have tiny plots, or tiny titles. You can see that the proposed version is already better than current one (http://scikit-image.org/docs/dev/auto_examples/edges/plot_line_hough_transform.html). This is the best version I managed to make. You could try to play with it by yourself :).

`plot_watershed.py`: there were not any changes for that file, but I've added them now. OK :)

sciunto commented:
Cool. I'm just waiting travis and then I merge, I promise :) There's already plenty of good changes.

Could you open a ticket for plot_ncut please? We should fix it before the next release.

soupault commented:
Yay! :fireworks: Thanks for reviews!
